### ISSUE TITLE
Swagger Integeration .NetcoreApp3.1

### ISSUE THREAD

**Issue by Aranir** [Cysharp/MagicOnion#226]
I created a sample app where I copied the basic example from the website (MyFirstService).

The unity integration works out of the box, but when I tried to add the Swagger integration, I just get a 

```
localhost is currently unable to handle this request.
HTTP ERROR 500
```

I added  the following packages:
- MagicOnion (2.6.3)
- MagicOnionHosting (2.6.3)
- MagicOnionHttpGateway (2.6.3)

Are there any other packages I need to add to have the swagger integration work properly?

I tested the swagger in the Sandbox application of the source, and it seemed to work flawlessly, but I noticed it uses .NetCoreApp 2.1.

Might it be an issue with the .NetCoreApp 3.1 which I currently have based my application on or might there be anything else missing?

If it would help I can also upload an example repo with minimal code, if that might help in pinpointing the problem.

The main code I used is:

```
class Program
{
    static async Task Main(string[] args)
    {
        var magicOnionHost = MagicOnionHost.CreateDefaultBuilder()
            .UseMagicOnion(
                new MagicOnionOptions(isReturnExceptionStackTraceInErrorDetail: true),
                new ServerPort("localhost", 12345, ServerCredentials.Insecure))
            .UseConsoleLifetime()
            .Build();

        var webHost = new WebHostBuilder()
            .ConfigureServices(collection =>
            {
                collection.AddSingleton<MagicOnionServiceDefinition>(magicOnionHost.Services.GetService<MagicOnionHostedServiceDefinition>().ServiceDefinition);
            })
            .UseKestrel()
            .UseStartup<Startup>()
            .UseUrls("http://localhost:5432")
            .Build();

        await Task.WhenAll(webHost.RunAsync(), magicOnionHost.RunAsync());
    }
}

public class Startup
{
    public void Configure(IApplicationBuilder app, MagicOnionServiceDefinition magicOnion)
    {

        app.UseMagicOnionSwagger(magicOnion.MethodHandlers, new SwaggerOptions("MagicOnion.Server", "Swagger Integration Test", "/")
        {
        });
        app.UseMagicOnionHttpGateway(magicOnion.MethodHandlers, new Channel("localhost:12345", ChannelCredentials.Insecure));
    }
}

}


	public interface IMyFirstService : IService<IMyFirstService>
	{
		UnaryResult<int> SumAsync(int x, int y);
	}

	public class MyFirstService : ServiceBase<IMyFirstService>, IMyFirstService
	{
		public async UnaryResult<int> SumAsync(int x, int y)
		{
			Logger.Debug($"Received:{x}, {y}");

			return x + y;
		}
	}
```

Any help would be appreciated.

**Comment by mayuki** [comment]
I tested MagicOnion.Hosting + MagicOnion.HttpGateway on .NET Core 3.0 (netcoreapp3.0) and it seems to work fine.

However, I found that it depends on ASP.NET Core version. If you are using ASP.NET Core 2.2, it works fine. If using ASP.NET Core 3.0 or later, it does not work properly.

**Comment by neuecc** [comment]
Ver 3.0.0 may solve it.

**Comment by ktanakaj** [comment]
Unfortunately, My app's Swagger Integeration is also not working by  MagicOnion 3.0.2 on netcoreapp3.1. It was worked by MagicOnion 2.1.2 on netcoreapp2.2.

**Comment by ktanakaj** [comment]
Hi! Our team found a solution for the Swagger Integration on netcoreapp3.1.

```
.ConfigureServices(collection =>
{
    collection.AddSingleton<MagicOnionServiceDefinition>(magicOnionHost.Services.GetService<MagicOnionHostedServiceDefinition>().ServiceDefinition);
    // add the following codes
    collection.Configure<KestrelServerOptions>(options =>
    {
        options.AllowSynchronousIO = true;
    });
})
```

It is working fine. Please check it and update README.md.

**Comment by neuecc** [comment]
thanks!
I've confirmed this bug and fixed in `3.0.8`, all code uses asynchronous only.
